firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/AabUpdater.java

```diff
@@ -46,12 +49,17 @@
   private final Object updateAabLock = new Object();
 
   AabUpdater() {
-    this(FirebaseAppDistributionLifecycleNotifier.getInstance());
+    this(FirebaseAppDistributionLifecycleNotifier.getInstance(), new HttpsUrlConnectionFactory(), Executors.newSingleThreadExecutor());
```

Comment:
Interested to know the reasoning for passing in an executor and having the HttpsUrlConnectionFactory class. Seems like the executor is only used once to put it in a variable. Is this providing more consistency in the tests?

Reply:
Yeah exactly! In both cases, the practical reason is that it makes testing easier. It allows the tests to inject an instance that the test has control over. In the case of `HttpsUrlConnectionFactory`, the test gets to inject a mock. (Previously we were mocking a package-private method of the class under test, which isn't ideal.) In the case of the `Executor`, the test gets to call `.awaitTermination()` so that it can be sure that all tasks have completed before asserting things about the resulting state.

Just in general though, classes should require their dependencies be passed in the constructor instead of instantiating them themselves. That way they don't encapsulate the functionality of their dependencies, and can instead be tested in isolation from them. See https://effectivejava.googleplex.com/third-edition#page=41